.github/workflows/test.yml

```
@@ -162,3 +162,17 @@ jobs:
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
       - name: Image digest
         run: echo ${{ steps.docker_build.outputs.digest }}
+
+  autodeploy:
+    runs-on: ubuntu-latest
+    needs: [docker-publish]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
```

This job has a step where it only executes if `github.ref == 'refs/heads/main'` so this job would be triggered for other events that would not execute any steps.

I would do the following:

```suggestion
    github.event_name == 'push' && github.ref == 'refs/heads/main'
```

and remove the `if` condition from the `Deploy Staging` step.